overlord/snapstate/snapstate.go

```diff
@@ -361,7 +365,7 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		prev = unlink
 	}
 
-	if !release.OnClassic && !snapsup.Flags.TestingOnlyLeaveOutKernetUpdateGadgetAssets && (snapsup.Type == snap.TypeGadget || snapsup.Type == snap.TypeKernel) {
+	if !release.OnClassic && !TestingLeaveOutKernetUpdateGadgetAssets && (snapsup.Type == snap.TypeGadget || snapsup.Type == snap.TypeKernel) {
```

Comment:
we want this to affect only the kernel, not the gadget I think. so it needs to be moved inside the parenthesis at the right place